pkg/names/generate.go

```diff
@@ -46,9 +50,16 @@ const (
 	maxGeneratedNameLength = maxNameLength - randomLength
 )
 
-func (simpleNameGenerator) GenerateName(base string) string {
+func (simpleNameGenerator) RestrictLengthWithRandomSuffix(base string) string {
+	if len(base) > maxGeneratedNameLength - 1 {
+		base = base[:maxGeneratedNameLength - 1]
+	}
+	return fmt.Sprintf("%s-%s", base, utilrand.String(randomLength))
+}
+
+func (simpleNameGenerator) RestrictLength(base string) string {
 	if len(base) > maxGeneratedNameLength {
 		base = base[:maxGeneratedNameLength]
```

Comment:
Why `maxGeneratedNameLength` instead of `maxNameLength`? Callers adding suffixes should be using `RestrictLengthWithRandomSuffix`, so the return value of this function could go all the way up to 63 chars, right?

Reply:
Honestly, oversight. Fixed it. =)